src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -100,11 +107,13 @@ private static JsonObject CreatePaginationConnection(JsonElement root, Paginatio
 
             if (paginationMetadata.RequestedEndCursor)
             {
-                // parse *Connection.endCursor if there are no elements
-                // if no after is added, but it has been requested HotChocolate will report it as null
-                if (returnedElemNo > 0)
+                // Note: if we do not add endCursor to the connection but it was in the request, its value will
+                // automatically be populated as null.
+                // Need to validate we have an extra element, because otherwise there is no next page
+                // and endCursor should be left as null.
+                if (returnedElementCount > 0 && hasExtraElement)
```

Comment:
Does `hasExtraElement` imply `returnedElementCount > 0`?
Because
```csharp
hasExtraElement = returnedElementCount == paginationMetadata.Structure!.Limit();
```

Reply:
actually, i suppose `returnedElementCount` could get to zero via the code:

```csharp
            if (isPaginationRequested)
            {
                // structure.Limit() is first + 1 for paginated queries where hasNextPage or endCursor is requested
                hasExtraElement = returnedElementCount == paginationMetadata.Structure!.Limit();
                if (hasExtraElement)
                {
                    // In a pagination scenario where we have an extra element, this element
                    // must be removed since it was only used to determine if there are additional
                    // records after those requested.
                    rootEnumerated = rootEnumerated.Take(rootEnumerated.Count() - 1);
                    --returnedElementCount;
                }
            }
```